### DISCUSSION TITLE
GPS is needed

### DISCUSSION THREAD

**Discussion by ks0002544** [tesla-android/issue-tracker#220] [General]
I use android smarthone as USB modem connected to pi4. But I can not use waze app because GPS is needed.  Is it possible to use my smartphone as GPS tracker for waze app?

**Comment by mikegapinski** [comment]
GPS will be forwarded from the car to tesla android via the browser. Writing a custom driver for android to consume location data

**Reply by m0nk89** [reply]
Thank you for all your work on this project!

In the meantime, would any RPi gps modules work with your version of Android plug and play?

**Reply by mikegapinski** [reply]
Nope, at least I did not test any

**Reply by yerode** [reply]
Great idea to use the car-gps-data! 👍

**Reply by yerode** [reply]
Could you give us an estimated timeframe for GPS support?

Curiously I tried an ublox 7 based GPS via USB. Doesn't work. ¯\_(ツ)_/¯

**Reply by mikegapinski** [reply]
It won't work when there is no driver present for it in the system :) This was my first idea, but I've figured out a way to do it without it and retuned all modules I purchased.

1 st rule of free and open source software, never ask for ETAs ;)

**Reply by yerode** [reply]
Thanks for your reply. 👍 I'll never ask again but I am looking forward to GPS support.

**Reply by UziKhana** [reply]
> GPS will be forwarded from the car to tesla android via the browser. Writing a custom driver for android to consume location data

Just staring out with Tesla Android, I am on MYP 2023 and just got updated to 2023.6.11 that has some browser updates, on my install no apps seem be getting location data from the browser. Waze or Google maps dont know it.

**Reply by mikegapinski** [reply]
> GPS will be forwarded

It's not fully wrapped yet, the current release does not have this feature. I am hoping to get it done soon, Google Maps and other navigation apps work flawlessly with the new multitouch driver
